testing/sync-test/src/auth.rs

```diff
@@ -165,19 +165,39 @@ impl TestAccount {
             &client_id,
             &scope,
             &jwk,
-            &self.auth_key,
+            &auth_key,
             &self.cfg,
             (&self.k_sync, &self.xcs),
         )?;
-        let auth_params = auth::AuthorizationParameters::new(
-            client_id.clone(),
-            code_challenge.clone(),
-            code_challenge_method.clone(),
-            scope.clone(),
-            keys_jwe,
-            state.clone(),
-        );
-        auth::send_authorization_request(&self.cfg, auth_params, &self.auth_key)
+        let auth_params = auth::AuthorizationRequestParameters {
+            client_id: client_id.clone(),
+            code_challenge: Some(code_challenge.clone()),
+            code_challenge_method: Some(code_challenge_method.clone()),
+            scope: scope.clone(),
+            keys_jwe: Some(keys_jwe),
+            state: state.clone(),
+            access_type: "offline".to_string(),
+        };
+        auth::send_authorization_request(&self.cfg, auth_params, &auth_key)
+    }
+
+    fn execute_oauth_pair_flow(&self, oauth_uri: &str) -> Result<(String, String)> {
+        let url = Url::parse(&oauth_uri)?;
+        let auth_params = auth::AuthorizationParameters::from(url)?;
```

Comment:
Thanks for trying out the use of `From` here, I think it helped make this method more accessible.